### ISSUE TITLE
NVDA not reading the objects under the mouse cursor when mouse emulator is used.

### ISSUE THREAD

**Issue by danewhk** [nvaccess/nvda#7480]
Some of our clients use emulator freeware such as “JoyToKey” to allow joystick to mimic a mouse, so they can control the mouse cursor with a joystick. It is found that NVDA would not read aloud the object or text under the mouse cursor when such emulator is used to control the mouse cursor. Do you have any suggestion on how we can make NVDA works with these mouse emulators? 

Thanks so much!

System configuration:

NVDA version:  2017.2 Installed
This issue exists in all versions. 

Windows version:　Windows 7  6.1.7601

Name and version of other software in use when reproducing the issue:
JoyToKey (January 2016 version)
Website:　http://www.oneswitch.org.uk/2/I/JoyToKey/JoyToKey.htm

**Comment by bhavyashah** [comment]
Can NVDA's mouse tracking be made to support mouse emulators and joysticks? I have no clue, but @danewhk, could your clients try the Golden Cursor NVDA add-on and report whether it suffices their needs?

**Comment by danewhk** [comment]
Thanks for your advises @bhavyashah. We have tried to operate the mouse with the combo keys of Golden Cursor, however, NVDA would not read aloud the items under the mouse cursor. We then tried to enable "review cursor follows mouse movement" in NVDA but still doesn't work. Thanks so much for your kind assistance!

**Comment by bhavyashah** [comment]
@danewhk  Could you please presss NVDA+M and ensure that mouse tracking is enabled while you perform these tests? CC @josephsl (who may have further ideas regarding Golden Cursor) and @jcsteh for diagnostic assistance

**Comment by jcsteh** [comment]
@QChristensen, I seem to recall you did some research on this a while back due to another user report. If so, do you have anything to offer? Please disregard if I'm misremebering.

**Comment by Qchristensen** [comment]
I did indeed.  What I found was that if the emulator uses a mouse driver, it works fine.  An example of this is Unified Remote, which lets you use the touch screen on your mobile phone to control your mouse pointer on your PC: https://www.unifiedremote.com/

I did specifically talk to the JoyToKey people and they advised they are injecting the mouse pointer (using SendInput from memory) to move it.  Because it moves similarly to the way some of our object navigation commands work, we ignore it and our mouse tracking doesn't pick it up.  XPadder is another similar type of software which works the same way.  Both companies confirmed they have no plans to change to a "driver" style of software.

When I looked into it, (at the time, for an OT using game controllers / flight joysticks with clients with Cerebral Palsy) the best option I could find at the time, was that there are numerous input devices which are "mice" but which are shaped more like the joysticks they were using.  The only downside to these is that they are more expensive than a cheap flight controller and software interface.

**Comment by leonardder** [comment]
#8459 should fix this.
